patches/common/chromium/frame_host_manager.patch

```diff
@@ -50,7 +50,7 @@ index 2314c6d1d29ebc61d5156644617e9eb088df5d43..9c75367a173b836027b3b7628b49b363
 +        overriden_site_instance =
 +            candidate_site_instance
 +                ? candidate_site_instance
-+                : SiteInstance::CreateForURL(browser_context,
++                : current_site_instance->GetRelatedSiteInstance(
```

Comment:
I'm afraid that the fix to this problem is not this simple. As @zcbenz has already mentioned, we need to force the renderer process to restart upon navigating, to properly restart Node. This is the reason Electron is forcing an independent site instance in this case. #15821 fixed the window.opener problem for sandboxed renderers only, where Node is disabled and therefore there is no need to restart the process. I agree with @MarshallOfSound in that we should find a way to move towards the Chromium process model while properly restarting Node without forcing full process restarts.